Source/WebKit/UIProcess/WebPermissionControllerProxy.cpp

```diff
@@ -57,4 +71,42 @@ void WebPermissionControllerProxy::query(const WebCore::ClientOrigin& clientOrig
     webPageProxy->queryPermission(clientOrigin, descriptor, WTFMove(completionHandler));
 }
 
+WebPageProxy* WebPermissionControllerProxy::mostReasonableWebPageProxy(const WebCore::SecurityOriginData& topOrigin, WebCore::PermissionQuerySource source)
+{
+    WebPageProxy* webPageProxy = nullptr;
+
+    auto findWebPageProxy = [&topOrigin, &webPageProxy] (auto* processes) {
+        if (!processes)
+            return; 
+
+        for (auto& process : *processes) {
+            for (auto* potentialWebPageProxy : getPtr(process)->pages()) {
+                if (WebCore::SecurityOriginData::fromURL(URL { potentialWebPageProxy->currentURL() }) != topOrigin)
+                    continue;
+                // The most reasonable webPageProxy is the newest one (the one with the greatest identifier).
+                if (webPageProxy && webPageProxy->identifier().toUInt64() > potentialWebPageProxy->identifier().toUInt64())
```

Comment:
I think it would be nice to add the following member function to ObjectIdentifier.h:
```
bool operator>(const ObjectIdentifier& other)
{
    return toUInt64() > other.toUInt64();
}
```

Then we could simplify the check here.

Reply:
Fixed!